### PR TITLE
Implement global exception handling

### PR DESCRIPTION
**Pull request by junaidiqbalmoj**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines

Please remove this line and everything above and fill the following sections:

https://tools.hmcts.net/jira/browse/PUB-516

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/demo/errorhandling/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Template exception handler, that handles a custom DataStorageNotFoundException,
+     * and returns a 404 in the standard format.
+     * @param ex The exception that has been thrown.
+     * @param request The request made to the endpoint.
+     * @return The error response, modelled using the ExceptionResponse object.
+     */
+
+    @ExceptionHandler(DataStorageNotFoundException.class)
+    ResponseEntity<ExceptionResponse> handlePublicationNotFound(DataStorageNotFoundException ex, WebRequest request) {
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/demo/errorhandling/GlobalExceptionHandler.java:23]: Only minor, however the method name should be changed from handlePublication to something relevant for data management

File: src/main/java/uk/gov/hmcts/reform/demo/controllers/RootController.java [hunk 2]
```diff
@@ -25,4 +26,14 @@
     public ResponseEntity<String> welcome() {
         return ok("Welcome to spring-boot-template");
     }
+
+    /**
+     * Dummy endpoint, that demonstrates how the Global Exception handler can be used to capture
+     * and parse exceptions into a standard format.
+     * @return A ResponseEntity
+     */
+    @GetMapping("/StoreFile")
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/demo/controllers/RootController.java:35]: Due to it only being a Dummy endpoint, feel free to ignore this one :) However, REST conventions would indicate the endpoint name should be lower case, and potentially just '/file', if the endpoint is just around getting files

**Review by ChrisS1512 — COMMENTED** [review]
Looks good :)

I think the build is failing due to checkstyles not passing, so will need sorting before the merge

File: src/main/java/uk/gov/hmcts/reform/demo/errorhandling/GlobalExceptionHandler.java (hunk 1 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/demo/errorhandling/GlobalExceptionHandler.java]: Code has been updated.

File: src/main/java/uk/gov/hmcts/reform/demo/controllers/RootController.java (hunk 2 above)
Comment (junaidiqbalmoj) [reply src/main/java/uk/gov/hmcts/reform/demo/controllers/RootController.java]: Yes, you are right. I have changed the end point.

**Review by ChrisS1512 — APPROVED** [review]

**Review by johnoneill46 — APPROVED** [review]

**Pull request merged**